### PR TITLE
fix: override default image

### PR DESCRIPTION
**Pull request by Siafu**
Signed-off-by: Kambui Nurse <siafudev@gmail.com>

### CODE REVIEW

File: pkg/universe.dagger.io/x/solomon@dagger.io/spectral/lint.cue [hunk 1]
```diff
@@ -105,10 +105,14 @@ import (
 	container: docker.#Run & {
 		// Default container image can be overrided.
 		//   'spectral' must be installed and in the PATH.
-		input:              docker.#Image | *_buildDefaultImage.output
-		_buildDefaultImage: docker.#Pull & {
-			source: "stoplight/spectral"
+		input: docker.#Image
```
Comment (TomChv) [inline pkg/universe.dagger.io/x/solomon@dagger.io/spectral/lint.cue:108]: You could also create a definition `#Image` and do `input: *#Image | docker.#Image`

For instance:

```cue
#Image: {
   docker.#Pull & {
      source: "stoplight/spectral"
   }
}
```

**Review by TomChv — COMMENTED** [review]
Left a comment to simplify your PR :D

File: pkg/universe.dagger.io/x/solomon@dagger.io/spectral/lint.cue (hunk 1 above)
Comment (Siafu) [reply pkg/universe.dagger.io/x/solomon@dagger.io/spectral/lint.cue]: changes pushed

**Review by TomChv — DISMISSED** [review]
LGTM

File: pkg/universe.dagger.io/x/solomon@dagger.io/spectral/lint.cue (hunk 1 above)
Comment (jpadams) [reply pkg/universe.dagger.io/x/solomon@dagger.io/spectral/lint.cue]: @TomChv I don't understand this approach. It doesn't work for me:

image

Comment (TomChv) [reply pkg/universe.dagger.io/x/solomon@dagger.io/spectral/lint.cue]: OH true, that's should be 
```cue
_default: #Image 

input: _base.output
```

File: pkg/universe.dagger.io/x/solomon@dagger.io/spectral/lint.cue [hunk 2]
```diff
@@ -105,10 +111,8 @@ import (
 	container: docker.#Run & {
 		// Default container image can be overrided.
 		//   'spectral' must be installed and in the PATH.
-		input:              docker.#Image | *_buildDefaultImage.output
-		_buildDefaultImage: docker.#Pull & {
-			source: "stoplight/spectral"
-		}
+		input: *#Image | docker.#Image
```
Comment (TomChv) [inline pkg/universe.dagger.io/x/solomon@dagger.io/spectral/lint.cue:114]: I missed a detail in my review, my bad!
See : https://github.com/dagger/dagger/pull/2486#discussion_r884326402

**Comment by jpadams** [comment]
@Siafu I'm guessing you might have an update to this PR now.

**Comment by jpadams** [comment]
Talked with @Siafu and it's good to close this one now. He's using the universe package.

**Pull request closed without merging**